c/datasetjson.c

```diff
@@ -1719,6 +1720,111 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
 #endif /* __ZOWE_OS_ZOS */
 }
 
+static int getDSCB(char* datasetName, char* dscb, int bufferSize){
+  if (bufferSize < INDEXED_DSCB){
+    zowelog(NULL, LOG_COMP_DATASETJSON, ZOWE_LOG_WARNING, 
+            "DSCB of size %d is too small, must be at least %d", bufferSize, INDEXED_DSCB);
+    return 1;
+  }
+  Volser volser = {0};
+
+  DatasetName dsn = {0};
+  memcpy(dsn.value, datasetName, DATASET_NAME_LEN);
```

Comment:
A dataset name is not necessarily ```DATASET_NAME_LEN``` bytes. This may 0C4. And for DSCB it probably needs to be space padded.